nupic/research/frameworks/vernon/distributed/experiments/self_supervised_experiment.py

```diff
@@ -0,0 +1,58 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from nupic.research.frameworks.vernon.distributed import (
+    SupervisedExperiment as DistributedSupervisedExperiment,
+)
+from nupic.research.frameworks.vernon.experiments import (
+    SelfSupervisedExperiment as SelfSupervisedExperimentBase,
+)
+
+__all__ = ["SelfSupervisedExperiment"]
+
+
+class SelfSupervisedExperiment(
+    DistributedSupervisedExperiment, SelfSupervisedExperimentBase
+):
+    """
+    Distributed SelfSupervisedExperiment.
+    """
+
+    @classmethod
+    def create_unsupervised_sampler(cls, config, dataset):
+        return cls.create_train_sampler(config, dataset)
```

Comment:
In the non-distributed class, you have a `num_unsupervised_samples` param, why doesn't something analogous apply here?

Reply:
This is something that I'll leave off for now. I used the RandomSampler for the non-distributed self-supervised experiment, but the DistributedSampler doesn't allow for a num_samples argument and instead just chops up the dataset according to the number of GPU's. I guess it's possible to create a new RandomDistributedSampler class, but I wasn't sure if something already exists that subsamples and distributes across machines.